server/src/main/java/io/spine/server/tuple/EitherOf5.java

```diff
@@ -111,48 +111,109 @@ private EitherOf5(Message value, int index) {
      * Obtains the value of the first alternative.
      *
      * @throws IllegalStateException if a value of another type is stored instead.
-     * @return the stored value.
+     * @return the stored value
+     * @see #hasA()
      */
     @Override
     public A getA() {
         return get(this, 0);
     }
 
+    /**
+     * Tells whether {@code <A>} value is stored.
+     *
+     * @return {@code true} if the first alternative value is set, {@code false} otherwise
+     */
+    @Override
+    public boolean hasA() {
+        return index() == 0;
+    }
+
     /**
      * Obtains the value of the second alternative.
      *
      * @throws IllegalStateException if a value of another type is stored instead.
-     * @return the stored value.
+     * @return the stored value
+     * @see #hasB()
      */
     @Override
     public B getB() {
         return get(this, 1);
     }
 
+    /**
+     * Tells whether {@code <B>} value is stored.
+     *
+     * @return {@code true} if the second alternative value is set, {@code false} otherwise
+     */
+    @Override
+    public boolean hasB() {
+        return index() == 1;
+    }
+
     /**
      * Obtains the value of the third alternative.
      *
      * @throws IllegalStateException if a value of another type is stored instead.
-     * @return the stored value.
+     * @return the stored value
+     * @see #hasC()
      */
     @Override
     public C getC() {
         return get(this, 2);
     }
 
+    /**
+     * Tells whether {@code <C>} value is stored.
+     *
+     * @return {@code true} if the third alternative value is set, {@code false} otherwise
+     */
+    @Override
+    public boolean hasC() {
+        return index() == 2;
+    }
+
     /**
      * Obtains the value of the third alternative.
      *
      * @throws IllegalStateException if a value of another type is stored instead.
-     * @return the stored value.
+     * @return the stored value
+     * @see #hasD()
      */
     @Override
     public D getD() {
         return get(this, 3);
     }
 
+    /**
+     * Tells whether {@code <D>} value is stored.
+     *
+     * @return {@code true} if the fourth alternative value is set, {@code false} otherwise
+     */
+    @Override
+    public boolean hasD() {
+        return index() == 3;
+    }
+
+    /**
+     * Obtains the value of the fifth alternative.
+     *
+     * @throws IllegalStateException if a value of another type is stored instead.
+     * @return the stored value
+     * @see #hasE()
+     */
     @Override
     public E getE() {
         return get(this, 4);
     }
+
+    /**
+     * Tells whether {@code <E>} value is stored.
+     *
+     * @return {@code true} if the fifth alternative value is set, {@code false} otherwise
+     */
+    @Override
+    public boolean hasE() {
+        return index() == 4;
```

Comment:
Now, I think we need a class with constants shared across the package. Please consider. We have too many magic constants that are obvious, but still are “magic constants”. Can we have something like `IndexOf.E` for example?